app/views/posts/_form.html.erb

```diff
@@ -0,0 +1,22 @@
+<%= form_with(model: post, local: true) do |form| %>
+  <% if post.errors.any? %>
+    <div id="error_explanation">
+      <h2><%= pluralize(post.errors.count, "error") %> prohibited this post from being saved:</h2>
+
+      <ul>
+        <% post.errors.full_messages.each do |message| %>
+          <li><%= message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
```

Comment:
* [ ] **_There should be NO application logic in the views (no `Post.new`, `Comment.create`, `current_user.friends.find_by(...)`...)._**
* [ ] **_View-related logic can be encapsulated in View helpers (app/helpers/\*_helper.rb)_**
* [ ] **_There should be absolutely NO logic in views._**

Reply:
We don't know how to put an embedded ruby code into an 'rb' file. We just added our if condition to the helper and if it is not the thing you want, we would like to hear from you what you ask for us to do. Can you be more specific? As far as we understand there is no application logic here. Those are view related logic.

Reply:
We don't know how to render an HTML file in a normal rb file. Our logic as far as we understand is not an application logic but is a view logic. Still, however, if it is not the right way, can you help us and give us some examples of how to do it?